server/methods/followUser.js

```diff
@@ -0,0 +1,43 @@
+import { Meteor } from 'meteor/meteor';
+
+import { Users } from '../../app/models';
+import { settings } from '../../app/settings';
+
+
+Meteor.methods({
+	followUser(username) {
+		if (settings.get('Newsfeed_enabled')) {
+			// Update followers keys
+			let userObject = Users.findOneByUsername(username);
+			if ('followers' in userObject) {
+				const followersObject = userObject.followers;
+				const { _id } = Users.findOneByUsername(Meteor.user().username);
+				followersObject[_id] = '';
+				Users.update(userObject._id, { $set: { followers: followersObject } });
+			} else {
+				const followersObject = new Object();
+				const { _id } = Users.findOneByUsername(Meteor.user().username);
+				followersObject[_id] = '';
+				Users.update(userObject._id, { $set: { followers: followersObject } });
+			}
+
+			// Update following keys
+			userObject = Users.findOneByUsername(Meteor.user().username);
+			if ('following' in userObject) {
+				const followingObject = userObject.following;
```

Comment:
As the number of Following/Followers increase, it would increase in size of each user object considerably. As the `User` object is perhaps the most commonly fetched object for number of other purposes in all other areas of the application, we might want to structure our schema as described here. Thoughts? @bizzbyster @dtoth 
Check [this](https://stackoverflow.com/questions/14459194/mongodb-how-to-design-twitter-style-followers-following-relation-model-in-mongo)

Reply:
@kb0304 @fliptrail I am not an expert on the server side architecture but I tend to agree with Karan based on the link he sent. If you are still unsure we can try to get some inputs from others on the RC team. But in my opinion this seems like a good approach.